components/persistor/lib/knex/db.ts

```diff
@@ -791,55 +791,88 @@ module.exports = function (PersistObjectTemplate) {
+                            try {
+                                await knex.schema.table(tableName, async function (table) {
+                                    await table['drop' + type]([], name);
+                                });
+                            }
+                            catch(error) {
+                                    logTableChangesWarningMessage(columns, type, name, error);
+                            };
+                        }
+                        else {
+                            try {
+                                await knex.schema.table(tableName, async function (table) {
+                                    await table[type](columns, name);
+                                })
+                            } catch(error) {
+                                logTableChangesWarningMessage(columns, type, name, error);
+                            };
+                        }
                     }
-                    else
-                        table[type](columns, name);
-
-                }));
+                    catch(error) {
+                        if (error && error.message && error.message === 'index type can be only "unique" or "index"') {
+                            throw error;
+                        }
+    
+                        const logger = PersistObjectTemplate && PersistObjectTemplate.logger;
+                        if (logger) {
+                            logger.warn({
+                                module: 'db',
+                                function: 'applyTableChanges',
+                                error: error
+                            }, 'Unable to apply index changes for '+ tableName);
+                        }
+                    };
+                }
             }
 
-
-            return knex.transaction(function (trx) {
-                return trx.schema.table(tableName, function (table) {
-                    syncIndexesForHierarchy('delete', dbChanges, table);
-                    syncIndexesForHierarchy('add', dbChanges, table);
-                    syncIndexesForHierarchy('change', dbChanges, table);
-                }).catch(function (error) {
-                    if (error && error.message && error.message === 'index type can be only "unique" or "index"') {
-                        throw error;
-                    }
-                    const logger = PersistObjectTemplate && PersistObjectTemplate.logger;
-                    if (logger) {
-                        logger.warn({
-                            component: 'persistor',
-                            dbChanges: dbChanges,
-                            module: 'db',
-                            activity: 'synchronizeIndexes',
-                            error: error
-                        }, 'Unable to update index');
-                    }
-                });
-            });       
+            await syncIndexesForHierarchy('delete', dbChanges);
```

Comment:
I'm wondering if these need to be done in order or can we do a Promise.all, especially now that we are removing them from a single transaction.

Reply:
I syncIndexesForHierarchy doesnt throw anymore, then it can be in a promise.all

Reply:
done